docker-bake.hcl

```diff
@@ -1,29 +1,75 @@
-variable "REPO_NAME" {
+variable "IMAGE_NAME" {
     default = "dunglas/frankenphp"
 }
 
 group "default" {
-    targets = ["bullseye", "alpine"]
+    targets = ["bullseye", /*"buster", "alpine315",*/ "alpine316"]
+}
+
+group "bullseye" {
+    targets = ["bullseye-php-82"]
+}
+
+group "buster" {
+    targets = ["buster-php-82"]
+}
+
+group "alpine315" {
+    targets = ["alpine315-php-82"]
+}
+
+group "alpine316" {
+    targets = ["alpine316-php-82"]
 }
 
 target "common" {
     platforms = ["linux/amd64", "linux/arm64"]
```

Comment:
If we remove `platforms` will it just build the native platform? This may be desired since we can set the platform in the Github Action but when building locally there really isn't (usually) a reason to build cross-platform.

Reply:
Check `docs/docker.md` there are examples on do different kinds of bake runs.

- Run a target
- Build only for 1 architecture

Flag `--load` and setting 1 arch is needed to build images to your local as the whole build is one inside special container.

Building for multiple platforms (architecture) is kind of one point of bake. BUT as I said, you can make partial builds with flags. Good for e.g. local testing. So for example for me using M1 mac I could build only alpine image(s):

```
docker buildx bake -f docker-bake.hcl --pull --load --set "*.platform=linux/arm64" alpine316
```

<img width="614" alt="CleanShot 2022-10-31 at 15 46 52@2x" src="https://user-images.githubusercontent.com/1140272/199022832-41608f3f-404f-4a02-81a6-58029eee1a4b.png">

Reply:
I feel we now mix things.

E.g. `docker build ...` can be used just fine when you want to have one image locally.

Now we have basically max 4 variants coming out from bake (buster x 2 archs + alpine x 2 archs). Let's imagine when we need su make build for 3 latest supported PHP versions and maybe more distros than 2 and for arm and amd. That would be 12 or more variants. And then we start to see the benefits of bake.

For the bake `--load` and `--set "*.platform=something` are more like debugging purposes: load "loads" the created images to your host (as they are build inside that special builder) and with setting platform we just say "build it for my arch". It does not detect it as it's kind of not it's job (to be used like that).

So I would suggest using just plain old `docker build... target-i-wanna-test` when wanna test one image on locally and use the `docker buildx bake ...` when just wanting to test that all builds pass (before pushing to github and running workflows).

When using bake in CI, it uses QEMU emulation for being able to run multiarch builds (as GHA does not have arm64 based runner - only self-hosted runner can even have arm64 for now).

Sorry, long answer =)